tiatoolbox/models/engine/deep_feature_extractor.py

```diff
@@ -0,0 +1,264 @@
+"""Define DeepFeatureExtractor class."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import dask.array as da
+from typing_extensions import Unpack
+
+from tiatoolbox.models.dataset.dataset_abc import WSIStreamDataset
+from tiatoolbox.utils.misc import get_tqdm
+
+from .semantic_segmentor import SemanticSegmentor, SemanticSegmentorRunParams
+
+if TYPE_CHECKING:  # pragma: no cover
+    import os
+    from collections.abc import Callable
+    from pathlib import Path
+
+    import numpy as np
+    from torch.utils.data import DataLoader
+
+    from tiatoolbox.annotation import AnnotationStore
+    from tiatoolbox.models.engine.io_config import IOSegmentorConfig
+    from tiatoolbox.models.models_abc import ModelABC
+    from tiatoolbox.wsicore import WSIReader
+
+
+class DeepFeatureExtractor(SemanticSegmentor):
+    """Generic CNN Feature Extractor."""
+
+    def __init__(
+        self: DeepFeatureExtractor,
+        model: str | ModelABC,
+        batch_size: int = 8,
+        num_workers: int = 0,
+        weights: str | Path | None = None,
+        dataset_class: Callable = WSIStreamDataset,
+        *,
+        device: str = "cpu",
+        verbose: bool = True,
+    ) -> None:
+        """Initialize :class:`DeepFeatureExtractor`."""
+        super().__init__(
+            model=model,
+            batch_size=batch_size,
+            num_workers=num_workers,
+            weights=weights,
+            device=device,
+            verbose=verbose,
+        )
+        self.process_prediction_per_batch = False
+        self.dataset_class = dataset_class
+
+    def infer_wsi(
+        self: SemanticSegmentor,
+        dataloader: DataLoader,
+        save_path: Path,
+        **kwargs: Unpack[SemanticSegmentorRunParams],
+    ) -> dict[str, da.Array]:
+        """Perform model inference on a whole slide image (WSI).
+
+        This method processes a WSI using the provided DataLoader, merges
+        patch-level predictions into a full-resolution canvas, and returns
+        the aggregated output. It supports memory-aware caching and optional
```

Comment:
this docstring is still holdover from the semanticSegemntor; we aren't making any canvas here right?